solr/core/src/java/org/apache/solr/security/PublicKeyHandler.java

```diff
@@ -32,18 +32,22 @@
 public class PublicKeyHandler extends RequestHandlerBase {
   public static final String PATH = "/admin/info/key";
 
+  //This is an optimization for tests only
+  public static volatile CryptoKeys.RSAKeyPair REUSABLE_KEYPAIR ;
```

Comment:
Why not use the key-pair that's already on disk? See https://github.com/apache/lucene-solr/blob/master/solr/test-framework/src/java/org/apache/solr/SolrTestCaseJ4.java#L289-L290